PlayList.java

```diff
@@ -71,34 +100,83 @@ public int indexOf(String title) {
      *  is full, does nothing and returns false. Otherwise, inserts the track and
      *  returns true. */
     public boolean add(int i, Track track) {
-        //// replace the following statement with your code
-        return false;
+    
+        if(size==maxSize || i<0 || i>=size) //if list is full or the given index is negative or bigger than size
+        {
+            return false;
+        }
+
+        if((tracks==null || size==0) && maxSize!=size)
+        {
+            tracks[i]=track;
+            
+        }
+        else
+        {
+            if(i<size-1 && size!=maxSize)
+            {
+                for(int j=size; j>i; j--)
+                {
+                    tracks[j]=tracks[j-1];
+                }
+                tracks[i]=track;
+                
+            }
+        }
+        size++;
+        
+
+        return true;
     }
      
     /** Removes the track in the given index from this list.
      *  If the list is empty, or the given index is negative or too big for this list, 
      *  does nothing and returns -1. */
     public void remove(int i) {
-        //// replace this comment with your code
+        if(size!=0 && i>=0 && i<=size-1)
+        {
+            for(int j=i;j<size-1;j++)
+            {
+                tracks[j]=tracks[j+1];
+            }
+            tracks[size-1]=null;
+            size--;
+        }
     }
 
     /** Removes the first track that has the given title from this list.
      *  If such a track is not found, or the list is empty, or the given index
      *  is negative or too big for this list, does nothing. */
     public void remove(String title) {
-        //// replace this comment with your code
+        for(int i=0; i<size; i++)
+        {
+            if(tracks[i].getTitle()==title)
+            {
+                remove(i);
+            }
+        }
```

Comment:
Instead of searching with loop, use the existing method you've created.
```suggestion
        int index = indexOf(title);
        if (index > 0 && index < this.size)
        {
            remove(index);
        }
```